parsl/dataflow/flow_control.py

```diff
@@ -141,8 +151,94 @@ def close(self):
         self._thread.join()
 
 
+class Timer(object):
+    '''This timer is a simplified version of the FlowControl timer.
+    This timer does not employ notify events.
+
+    This is based on the following logic :
+
+    >>> BEGIN (INTERVAL, THRESHOLD, callback) :
```

Comment:
Same comment about the Python interpreter prompt.

Also, this is just personal preference, so no pressure to actually change this: I prefer to avoid toy code in documentation in most cases. The actual code this toy code is describing is only ~2x as long, and Python syntax is so simple it is basically toy code. But sometimes toy code cannot convey all of the proper details, so in the end you read the toy code and the real code and have to do the additional work of mapping between the two. So I prefer to use docs for exposition. Just my two cents.